pkg/storegateway/bucket_chunk_reader_test.go

```diff
@@ -91,3 +97,83 @@ func TestBucketChunkReader_refetchChunks(t *testing.T) {
 		})
 	}
 }
+
+func BenchmarkBucketChunkReader_loadChunks(b *testing.B) {
+	const chunkSize = 256
+
+	bcs := []int{
+		1000,
+		10_000,
+		100_000,
+		1_000_000,
+	}
+	for _, numChunks := range bcs {
+		b.Run(fmt.Sprintf("blocks-%d", numChunks), func(b *testing.B) {
+			chunkObjs := make([]string, numChunks)
+			for i := 0; i < len(chunkObjs); i++ {
+				chunkObjs[i] = fmt.Sprintf("chunk-%d", i)
+			}
+
+			ctx := context.Background()
+			block := &bucketBlock{
+				bkt:          newMockObjectStoreBucketReader(chunkSize),
+				partitioners: blockPartitioners{naivePartitioner{}, naivePartitioner{}, naivePartitioner{}},
+				chunkObjs:    chunkObjs,
+			}
+
+			reader := newBucketChunkReader(ctx, block)
+
+			// Prepare mock data for testing.
+			loadIdxs := make([]loadIdx, numChunks)
+			for i := 0; i < len(loadIdxs); i++ {
+				loadIdxs[i] = loadIdx{
+					offset:      uint32(i * chunkSize),
+					length:      chunkSize,
+					seriesEntry: 0,
+					chunkEntry:  i,
+				}
+			}
+
+			// Run the benchmark.
+			b.ReportAllocs()
+			b.ResetTimer()
+
+			for i := 0; i < b.N; i++ {
+				reader.reset()
+
+				for i, idx := range loadIdxs {
+					if err := reader.addLoad(chunks.ChunkRef(i), idx.seriesEntry, idx.chunkEntry, idx.length); err != nil {
+						b.Fatalf("error adding load: %v", err)
+					}
+				}
+				var res = make([]seriesChunks, 1)
+				res[0].chks = make([]storepb.AggrChunk, len(loadIdxs))
+
+				chunksPool := pool.NewSafeSlabPool[byte](chunkBytesSlicePool, seriesChunksSlabSize)
```

Comment:
i'm on the fence whether we should be including this in the benchmark. Technically the size of the chunks and how much memory the reader has to allocate for the chunks is outside of the control of `reader.load`. I think instantiating one pool for each benchmark case and using it for all `b.N` runs will only record the allocations that the chunksReader creates along the way (like the allocation you just removed in this PR)